build.cmd

```diff
@@ -6,8 +6,26 @@ setlocal
 ::       means that that rebuilding cannot successfully delete the task
 ::       assembly. 
 
-echo %2 | find /I "outerloop"
-if "%errorlevel%" equ "0" (
+set outloop=false
+
+:START_CMDLINE_PARSE
+set SWITCH=%1
+if {%SWITCH%} == {} goto :END_CMDLINE_PARSE
+set VALUE=%2
+
+if /i "%SWITCH%" == "/p:WithCategories" (
+  if /i "%VALUE%" == "OuterLoop" (
+      set outloop=true
+      goto :END_CMDLINE_PARSE
+  )
+)
+
+SHIFT
+SHIFT
```

Comment:
If an earlier argument is passed using an =, but then the /p:WithCategories argument is passed with a space, this will cause the /p:WithCategories to be missed. E.g. if the parameters are "/p:Coverage=true /p:WithCategories OuterLoop" then the double shift will shift the "/p:Coverage=true" and the "/p"WithCategories" values off the argument parameters leaving just "OuterLoop".